### PR TITLE
Update sinusoidal_positional_embedding.py

### PR DESCRIPTION
**Pull request by zherowolf**
fixing bugs with error "argument must be tuple of ints, not Tensor"

### CODE REVIEW

**Comment by jhcross** [comment]
@zherowolf, can you clarify which use of bsz or seq_len is causing the error?

**Comment by jhcross** [comment]
We can't merge this change as-is because it breaks ONNX export to Caffe2 (see, e.g., https://github.com/pytorch/translate/blob/0728a48757436015b04ab9c3073dc5d65819a9a2/pytorch_translate/test/test_onnx.py#L216), but I'll be happy to fix the error in another way if you can provide details of the line where it is occurring.

**Comment by zherowolf** [comment]
the bug occurs in [https://github.com/pytorch/fairseq/blob/master/fairseq/modules/sinusoidal_positional_embedding.py#L85](url)

**Comment by zherowolf** [comment]
the bug occurs in https://github.com/pytorch/fairseq/blob/master/fairseq/modules/sinusoidal_positional_embedding.py#L85

**Comment by jhcross** [comment]
Thanks @zherowolf ! And does it fix that error if instead of passing bsz and seq_len directly, you pass them through an `int` constructor:

`        return self.weights.index_select(0, positions.view(-1)).view(int(bsz), int(seq_len), -1).detach()
`

**Comment by zherowolf** [comment]
great resolution. That works fine.

**Comment by myleott** [comment]
Closing, since this seems resolved, thanks!
